pkg/services/publicdashboards/service/query.go

```diff
@@ -139,8 +136,9 @@ func (pd *PublicDashboardServiceImpl) GetQueryDataResponse(ctx context.Context,
 		return nil, models.ErrPanelQueriesNotFound.Errorf("GetQueryDataResponse: failed to extract queries from panel")
 	}
 
-	anonymousUser := buildAnonymousUser(ctx, dashboard, pd.features)
-	res, err := pd.QueryDataService.QueryData(ctx, anonymousUser, skipDSCache, metricReq)
+	// We don't have a signed in user for public dashboards. We are using Grafana's Identity to query the datasource.
+	svcCtx, svcIdent := identity.WithServiceIdentity(ctx, dashboard.OrgID)
+	res, err := pd.QueryDataService.QueryData(svcCtx, svcIdent, skipDSCache, metricReq)
```

Comment:
Technically this change is not needed as I don't think we rely on the external query service yet. I'm just aligning the behavior. Maybe I should revert. :thinking: